### DISCUSSION TITLE
How to stream RTSP from IP Cam to web page

### DISCUSSION THREAD

**Discussion by zebox** [bluenviron/mediamtx#2713] [Questions]
### Question

Could you explain how can I re-stream RTSP stream from IP camera (Hikvision) to web page? Which protocol should I use for show video on a web page with HTML5? Can I use docker image without ffmpeg for that or ffmpeg is required?

**Comment by zebox** [comment]
I tried to add path with `ffmpeg`

```text
{
  "name": "my_path",
  "runOnInit": "ffmpeg -i \"rtsp://mylogin:mypass@192.168.1.55/Streaming/Channels/101?transportmode=mcast&profile=Profile_1\" -c:v copy -tag:v hvc1 -an -hls_time 2 -hls_list_size 5 -hls_flags delete_segments+append_list -start_number 1 index.m3u8"
}
```
But when i try open `http://mediamtx.local:8888/my_path` i get error message in browser JS console 
`http://mediamtx.local:8888/my_path/index.m3u8 not found`

What do I wrong?

**Comment by aler9** [comment]
Hello, instructions on how to ingest a RTSP stream into the server are in the README:
https://github.com/bluenviron/mediamtx#rtsp-cameras-and-servers

instructions  on how to read that stream with a web browser are in the README:
https://github.com/bluenviron/mediamtx#web-browsers-1

No FFmpeg, API or anything else is required.

**Reply by webysther** [reply]
Hi, can u share where in documentation have the rtsp camera and servers tutorial?

Edit: I find it!

For hikvision:

```dotenv
MTX_RTSPTRANSPORTS=tcp
MTX_WEBRTCADDITIONALHOSTS=REVERSE_PROXY_IP
MTX_PATHS_ANYNAME_SOURCE=rtsp://admin:pass@IP_CAMERA:554/Streaming/Channels/101
```

and

```html
<iframe src="https://example.com/anyname/" scrolling="no"></iframe>
```

**Comment by zebox** [comment]
Thanks for reply. My issue was in  the h.265 codec (no streaming to hls or to webrtc). I solved it when switched codec to h.264 in the camera settings.

**Reply by aler9** [reply]
@zebox H265 is supported by HLS indeed, you just have to use Chrome on Windows

[a comment by an automated account is not shown]
